src/components/dataset/schemaCreation/DatasetSchemaBuilderView.tsx

```diff
@@ -0,0 +1,780 @@
+      backgroundColor: '#272822',
+      borderRadius: 10,
+      padding: 20,
+    },
+  } as any);
+
+const DatasetSchemaBuilderView = withStyles(styles)(({ classes }: any) => {
+  const { register, getValues, setValue } = useFormContext();
+  const [datasetSchema, setDatasetSchema] = useState(
+    getValues().schema as DatasetSpecificationModel,
+  );
+  const [selectedTable, setSelectedTable] = useState(-1);
+  const [selectedColumn, setSelectedColumn] = useState(-1);
+  const [expandedTables, setExpandedTables] = useState({} as any);
+  const [relationshipModalOpen, setRelationshipModalOpen] = useState(false);
+  const [relationshipModalDefaultValues, setRelationshipModalDefaultValues] = useState({
+    from: '',
+    to: '',
+    expandedTables: {},
+  });
+
+  const [anchorElDetailsMenu, setAnchorElDetailsMenu] = useState<null | HTMLElement>();
+  const openDetailsMenu = Boolean(anchorElDetailsMenu);
+
+  const handleClickDetailsMenu = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorElDetailsMenu(event.currentTarget);
+  };
+  const handleCloseDetailsMenu = () => setAnchorElDetailsMenu(null);
+  type DetailsMenuButtonProps = {
+    children: React.ReactNode;
+  };
+  function DetailsMenuButton(props: DetailsMenuButtonProps) {
+    const { children } = props;
+    return (
+      <>
+        <IconButton
+          id="details-menu-button"
+          size="small"
+          color="primary"
+          className={classes.iconButton}
+          onClick={handleClickDetailsMenu}
+          aria-controls={openDetailsMenu ? 'details-menu' : undefined}
+          aria-haspopup="true"
+          aria-expanded={openDetailsMenu ? 'true' : undefined}
+        >
+          <MoreVert />
+        </IconButton>
+        <Menu
+          id="details-menu"
+          anchorEl={anchorElDetailsMenu}
+          open={openDetailsMenu}
+          onClose={handleCloseDetailsMenu}
+          MenuListProps={{
+            'aria-labelledby': 'details-menu-button',
+          }}
+        >
+          {children}
+        </Menu>
+      </>
+    );
+  }
+
+  const onJsonViewerChange = useCallback(
+    (value: string) => {
+      try {
+        const potentialSchema = JSON.parse(value);
+        // setDatasetSchema(potentialSchema);
+        setValue('schema', potentialSchema);
+      } catch (e) {
+        // do nothing
+      }
+    },
+    [setValue],
+  );
+
+  const createTable = () => {
+    if (!datasetSchema.tables) {
+      datasetSchema.tables = [];
+    }
+    const newIndex = datasetSchema.tables.length;
+    setDatasetSchema({
+      ...datasetSchema,
+      tables: [
+        ...datasetSchema.tables,
+        {
+          name: `table_name${datasetSchema.tables.length || ''}`,
+          columns: [],
+          primaryKey: [],
+        },
+      ],
+    });
+    setSelectedTable(newIndex);
+    setSelectedColumn(-1);
+    setExpandedTables({
+      ...expandedTables,
+      [newIndex]: true,
+    });
+  };
+
+  const deleteTable = () => {
```

Comment:
I mentioned the idea of having a confirmation dialog to confirm before deleting. I don't think I see that yet. 
(Also totally fine to push that off to another PR/ticket)